rosetta-ide/src/main/java/com/regnosys/rosetta/ide/quickfix/RosettaQuickFixCodeActionService.java

```diff
@@ -43,29 +54,79 @@ public class RosettaQuickFixCodeActionService implements ICodeActionService2 {
 
 	@Inject
 	private IQuickFixProvider quickfixes;
+	@Inject
+	private RangeUtils rangeUtils;
+	@Inject 
+	private ImportManagementService importManagementService;
 
 	@Override
 	public List<Either<Command, CodeAction>> getCodeActions(Options options) {
 		boolean handleQuickfixes = options.getCodeActionParams().getContext().getOnly() == null
 				|| options.getCodeActionParams().getContext().getOnly().isEmpty()
 				|| options.getCodeActionParams().getContext().getOnly().contains(CodeActionKind.QuickFix);
 
-		if (!handleQuickfixes) {
-			return Collections.emptyList();
-		}
-
 		List<Either<Command, CodeAction>> result = new ArrayList<>();
-		for (Diagnostic diagnostic : options.getCodeActionParams().getContext().getDiagnostics()) {
-			Options diagnosticOptions = createOptionsForSingleDiagnostic(options, diagnostic);
-			List<DiagnosticResolution> resolutions = quickfixes.getResolutions(diagnosticOptions, diagnostic).stream()
-					.sorted(Comparator.nullsLast(Comparator.comparing(DiagnosticResolution::getLabel)))
-					.collect(Collectors.toList());
-			for (DiagnosticResolution resolution : resolutions) {
-				result.add(Either.forRight(createFix(resolution, diagnostic)));
+		if (handleQuickfixes) {
+			for (Diagnostic diagnostic : options.getCodeActionParams().getContext().getDiagnostics()) {
+				Options diagnosticOptions = createOptionsForSingleDiagnostic(options, diagnostic);
+				List<DiagnosticResolution> resolutions = quickfixes.getResolutions(diagnosticOptions, diagnostic).stream()
+						.sorted(Comparator.nullsLast(Comparator.comparing(DiagnosticResolution::getLabel)))
+						.collect(Collectors.toList());
+				for (DiagnosticResolution resolution : resolutions) {
+					result.add(Either.forRight(createFix(resolution, diagnostic)));
+				}
 			}
 		}
+		
+		// Handle Sorting Imports
```

Comment:
Hm, could you show me how this works (e.g., in VS Code)? Is there a unit test?

Reply:
I am a bit surprised to see this here - since I thought a code action was something different as a quick fix. But they seem to be the same. So I wanna know what it looks like. :)

Reply:
This does make me wonder about responsibilities. Right now, this class is specifically called `...QuickFixCodeActionService`. Should we rename it? Or put it in a different class perhaps?

Reply:
Right now, this class was purely meant as a patch on top of Xtext's class. If we're adding functionality to it, I think we should restructure things, otherwise it's gonna get messy.